cfme/infrastructure/virtual_machines.py

```diff
@@ -302,12 +302,14 @@ class InfraVmReconfigureView(BaseLoggedInPage):
     cpu_total = WInput()  # read-only, TODO widgetastic
 
     disks_table = DisksTable()
+    affected_vms = Table('.//div[@id="records_div" or @id="miq-gtl-view"]//table')
 
     submit_button = Button('Submit', classes=[Button.PRIMARY])
     cancel_button = Button('Cancel', classes=[Button.DEFAULT])
 
-    # The page doesn't contain enough info to ensure that it's the right VM -> always navigate
-    is_displayed = False
+    @property
+    def is_displayed(self):
+        return self.context['object'].name in [row.name.text for row in self.affected_vms.rows()]
```

Comment:
**Required**

This check only verifies the current VM is in the row, but not that its the only one. I think this needs a more strict check that the given VM object is the _only_ VM in the view.

Reply:
As an example, the browser could actually be on the VM collection page, and the `affected_vms` widget is going to locate the table with the VMs. The name could be in the current page, and this check would return True, even though we aren't on the correct page at all.

Reply:
Updated.